diesel/src/pg/expression/array.rs

```diff
@@ -0,0 +1,103 @@
+use std::marker::PhantomData;
+use expression::{AppearsOnTable, Expression, IntoSingleTypeExpressionList, SelectableExpression};
+
+#[derive(Debug)]
+pub struct Array<T, ST> {
+    elements: T,
+    _marker: PhantomData<ST>,
+}
+
+/// Creates an `ARRAY[...]` expression.  The argument should be a tuple of
+/// expressions which can be represented by the same SQL type.
+///
+/// If the type can't be inferred, call `array::<SomeSqlType, _>((...))` to
+/// specify it.
+///
+/// # Examples
+///
+/// ```rust
+/// # #[macro_use] extern crate diesel;
+/// # include!("../../doctest_setup.rs");
+/// #
+/// # table! {
+/// #     users {
+/// #         id -> Integer,
+/// #         name -> VarChar,
+/// #     }
+/// # }
+/// #
+/// # #[cfg(feature = "postgres")]
```

Comment:
Because this lives in `diesel/src/pg` I think we  do not need the `#[cfg(feature = "postgres")]` annotation.

(Same for the empty main function for the not-pg case)

Reply:
:+1: